qa/src/test/java/io/camunda/zeebe/process/test/qa/ExtendAbstractsTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2021 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.process.test.qa;
+
+import com.google.common.reflect.ClassPath;
+import io.camunda.zeebe.process.test.extension.ZeebeProcessTest;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+class ExtendAbstractsTest {
+
+  public static final String EMBEDDED = "embedded";
+  public static final String TESTCONTAINER = "testcontainer";
+  public static final Class<ZeebeProcessTest> EMBEDDED_ANNOTATION = ZeebeProcessTest.class;
+  public static final Class<io.camunda.zeebe.process.test.extension.testcontainer.ZeebeProcessTest>
+      TESTCONTAINER_ANNOTATION =
+          io.camunda.zeebe.process.test.extension.testcontainer.ZeebeProcessTest.class;
+  private static final String BASE_PACKAGE = "io.camunda.zeebe.process.test.qa";
+  private static final String ABSTRACTS = "abstracts";
+
+  @Test
+  void testAbstractClassesAreExtendedWithBothExtensions() throws IOException {
+    final Map<String, List<Class<?>>> classes = new HashMap<>();
+    classes.put(ABSTRACTS, new ArrayList<>());
+    classes.put(EMBEDDED, new ArrayList<>());
+    classes.put(TESTCONTAINER, new ArrayList<>());
+
+    ClassPath.from(ClassLoader.getSystemClassLoader()).getTopLevelClasses().stream()
```

Comment:
:wrench: This seems correct, but also hard to read. I would be strongly in favor of using something like https://github.com/ronmamo/reflections or https://github.com/classgraph/classgraph. It's maybe overkill for what you are trying to achieve. But my hope is that it would be easier to comprehend. And you could also make the tests more precise, e.g. abstract classes should be extended, and they should have a certain annotation on them, or something like that.